packages/fleet/lib/fleet.ts

```diff
@@ -1,19 +1,89 @@
+import { Err, Ok } from '@nangohq/utils';
 import type { Result } from '@nangohq/utils';
 import { DatabaseClient } from './db/client.js';
 import * as deployments from './models/deployments.js';
-import type { CommitHash, Deployment } from './types.js';
+import * as nodes from './models/nodes.js';
+import type { CommitHash, Deployment, Node, RoutingId } from './types.js';
+import { FleetError } from './utils/errors.js';
+import { setTimeout } from 'node:timers/promises';
+import { Supervisor } from './supervisor.js';
+import type { NodeProvider } from './node-providers/node_provider.js';
+import type { FleetId } from './instances.js';
+import { noopNodeProvider } from './node-providers/noop.js';
+import { envs } from './env.js';
+
+const defaultDbUrl =
+    envs.NANGO_DATABASE_URL ||
+    `postgres://${encodeURIComponent(envs.NANGO_DB_USER)}:${encodeURIComponent(envs.NANGO_DB_PASSWORD)}@${envs.NANGO_DB_HOST}:${envs.NANGO_DB_PORT}/${envs.NANGO_DB_NAME}`;
```

Comment:
We are going to use a dedicated DB for that or just a different schema?
because for the keystore we choose a different path of setting migrations in a different schema but using the same schema for the table

Reply:
Right now it is setup to use different schema for the tables and the migrations. My reasoning is that fleet is somehow independent from the rest of the codebase, both codewise and conceptually. Wdyt?

Reply:
Ideally infra is not directly tied to the code but since it's now a first-citizen I'm not sure the project will be usable without this, aka not independent from the rest of the code. So I wouldn't bother splitting. But doesn't change much, as you wish